vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -653,6 +670,38 @@ public void setReceiver(Receiver receiver) {
         } else {
             setMaxFiles(1);
         }
+
+        runBeforeClientResponse(ui -> getElement().setAttribute("target",
+                new StreamReceiver(getElement().getNode(), "upload",
+                        getStreamVariable())));
+    }
+
+    /**
+     * Set the upload handler to be used for this upload component.
+     * <p>
+     * The given handler defines how uploaded file content is handled on the
+     * server and invoked per each single file to be uploaded. Note! This method
+     * overrides the receiver set by {@link #setReceiver(Receiver)} and removes
+     * the maximum files limit.
```

Comment:
You could set max files after setting the handler, which would still configure it.

Let's also mention in `setMaxFiles` that this is not effective when using a handler, specifically that there is no server-side check for it, and that max files are only checked on the client side..

Reply:
> You could set max files after setting the handler, which would still configure it.

Yes, definitely.

> Let's also mention in setMaxFiles that this is not effective when using a handler, specifically that there is no server-side check for it, and that max files are only checked on the client side.

Makes sense, I'll add this note.

Reply:
Actually this
```
specifically that there is no server-side check for it
```
should remain the same as for `Receiver`.

this check 
```
        if (getMaxFiles() != 0 && getMaxFiles() <= activeUploads) {
            throw new IllegalStateException(
                    "Maximum supported amount of uploads already started");
        }
```
should be performed with `UploadHandler` as well.

Reply:
added a small note in javadoc